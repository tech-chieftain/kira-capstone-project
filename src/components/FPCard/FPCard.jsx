import React from "react";
import PropTypes from "prop-types";
import Card from "react-bootstrap/Card";
import styled from "styled-components";
import SmallProfileCard from "../SmallProfileCard/SmallProfileCard";

const Container = styled(Card)`
  height: 415px;
  width: 292px;
  border: none:
  padding: 0;
  margin: 20px 5px;

  .img-container {
    height: 415px;
    width: 292px;
    object-fit: cover;
  }

  .title {
    margin-top: -22.5rem;
    font-size: 26px;
  }

`;

const FPCard = ({ overlay, width, imgSrc, title, subtitle, profilePicture, imgHeight }) => {
  if (overlay)
    return (
      <Container className="text-white container border-0 ">
        <Card.Img variant="top" src={imgSrc} alt="" className="img-container" />
        <Card.ImgOverlay className="mx-2">
          <Card.Subtitle>{subtitle}</Card.Subtitle>
          <Card.Title className="title">{title}</Card.Title>
        </Card.ImgOverlay>
      </Container>
    );

  return (
    <Card className="second-container">
      <Card.Img variant="top" src={imgSrc} alt="" />
      <Card.Body>
        <SmallProfileCard profilePicture={profilePicture} name={title} job={subtitle} />
      </Card.Body>
    </Card>
  );
};

FPCard.propTypes = {
  overlay: PropTypes.bool,
  width: PropTypes.number,
  title: PropTypes.string,
  subtitle: PropTypes.string,
  imgSrc: PropTypes.string,
  imgHeight: PropTypes.number,
  profilePicture: PropTypes.string,
};

FPCard.defaultProps = {
  overlay: false,
  // width: 18,
  title: "",
  subtitle: "",
  imgSrc: "https://picsum.photos/200/300",
  // imgHeight: 20,
};

export default FPCard;
