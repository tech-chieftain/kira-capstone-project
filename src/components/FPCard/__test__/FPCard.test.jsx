import * as React from "react";
import * as renderer from "react-test-renderer";
import FPCard from "../FPCard";

it("renders FPCard with text overlay correctly", () => {
  const tree = renderer.create(<FPCard />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders FPCard with caption correctly", () => {
  const tree = renderer.create(<FPCard overlay={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});
