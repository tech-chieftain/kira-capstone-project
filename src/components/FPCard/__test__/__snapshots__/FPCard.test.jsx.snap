// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FPCard with caption correctly 1`] = `
<div
  className="card"
  style={
    Object {
      "width": "18rem",
    }
  }
>
  <img
    alt=""
    className="card-img-top"
    src="https://picsum.photos/200/300"
  />
  <div
    className="card-body"
  >
    <div
      className="card"
      style={
        Object {
          "border": "0",
          "width": "16rem",
        }
      }
    >
      <div
        className="card-body"
      >
        <div
          className="align-items-center row"
        >
          <div
            className="col-4"
          >
            <div
              className="d-flex justify-content-end"
            >
              <svg
                fill="currentColor"
                height="3em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 496 512"
                width="3em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
                />
              </svg>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="card-title h5"
            >
              
            </div>
            <div
              className="card-subtitle h6"
            >
              
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders FPCard with text overlay correctly 1`] = `
<div
  className="text-white card"
  style={
    Object {
      "width": "18rem",
    }
  }
>
  <img
    alt=""
    className="card-img-top"
    src="https://picsum.photos/200/300"
  />
  <div
    className="card-img-overlay"
  >
    <div
      className="card-subtitle h6"
    >
      
    </div>
    <div
      className="card-title h5"
    >
      
    </div>
  </div>
</div>
`;
